### ISSUE TITLE
Space mining via cargo transmat

### ISSUE THREAD

**Issue by Deadlock989** [Deadlock989/IndustrialRevolution#251]
Experimental add-on mod, not sure if it will go anywhere, reached about 90% polish at time of writing.

![quantum-satellite](https://github.com/Deadlock989/IndustrialRevolution/assets/3128845/8577d74e-e024-4696-86f9-f435048648d8)

Launch a "quantum mining probe" in a rocket instead of a regular satellite and your daily (every 25000 ticks) cargo transfers will increase. 1000 ore returned per probe launched (split between five ore types), up to an effective maximum of 500 launches or 125 cargo transmats to cap potential UPS drag. Cargo transmats use a flat 200MJ per connection (the same as a regular transmat) regardless of ore amount returned. The four main ores plus "comet ice" (easily processed in natural gas and carbon dioxide) are returned in biased quantities (tilted towards iron and copper ore), but individual transmat inventories can be filtered, allowing unwanted resources to be ignored. The costs involved are astronomical (literally) compared to running a single advanced mining drill which can return roughly three times as much ore - but the advantage is that a cargo transmat can be placed anywhere you like and never runs out.

Would you use it?

https://github.com/Deadlock989/IndustrialRevolution/assets/3128845/a310750b-c4e2-415f-8b4f-a98bbb2074ba

**Comment by Cadams111** [comment]
I would absolutely install it, and start a brand new world of IR3 with the next set of balance changes to go along. I’m not sure if this was a factor, but I remember requesting something like this a while ago, so of course I would

**Comment by areyesca** [comment]
YES! add it, will be a nice resource when you are in the end-game. Any alternative use to rockets will be motivating!

**Comment by Deadlock989** [comment]
Still playing with this and reworking it. I don't like composite entities - they are always somehow lacking. In this case I needed something with the energy buffer behaviour of IR3's existing transmats but combined with the storage capacity of a chest. No good candidate exists so I was using a composite entity (EEI + hidden chest). The interface was lacking (missing entity preview for invisible chest, having to bolt on some weird relative GUI thing), it was hard to catch copy-paste settings events for the inventory because EEIs have restricted gui modes, there were various copy-paste and blueprint shenanigans required, etc. etc. Just horrible inconsistent behaviour and a ton of code that will inevitably get broken by some other mod's shenanigans.

So I gave up and made the cargo transmat just "lock on" to chests that touch it. Much less work to track new/deleted stuff, all the existing filtering copy-paste mechanics just work, allows for much clearer custom GUI on the cargo transmat, multiple transmats can share one inventory, you can choose to have bots fetch from the transmat dumps by using logistics containers, etc. etc. etc. Just better.

![image](https://github.com/Deadlock989/IndustrialRevolution/assets/3128845/8197fc02-49c7-44d4-8281-071603138d4d)

**Comment by Deadlock989** [comment]
Further balancing, something still niggling me but I can't put my finger on it.

Below represents the maximum number of container slots you can attach to a cargo transmat with IR3 only (you can get a lot more if you install a warehousing mod). Capping maximum transmats at 100 keeps UPS nice and manageable (it could go somewhat higher than that and still be OK but the lower the better). I increased the daily haul per probe launched to 1600 ore daily - that's close to one advanced mining drill running continuously for the day without productivity research or beacons. This means that if you reach a maximum of 1000 probes launched, and max out your transmats to 100, you could take delivery of the whole thing with just 4 steel chests per transmat.

GUI is very much WIP but shows the current breakdown of ores (as a percentage of the daily 1600). The last light blue one is comet ice, which produces mostly natural gas and some carbon dioxide when run through a chemical plant. The dark one is "carbonaceous ore" which is currently crushable into coke and gravel, but I'm not happy with that. I experimented with returning IR1's crushed carbon gravel from coal as a substitute for the coke but it just got messy. Maybe I should just go with coal, I know space coal isn't exactly realistic but this is a fantasy video game, not a simulator.

![image](https://github.com/Deadlock989/IndustrialRevolution/assets/3128845/6eb35328-c2c2-40fb-a2b5-d5af365a8375)

**Comment by Cadams111** [comment]
What are those blue star chests? Also are we getting an infinity chest reskin? I really appreciate your graphics work and making it all look good

**Comment by Deadlock989** [comment]
Those are just steel chests. I had to tweak chest sprites to make them line up better with loaders, so while I was in there I gave them the standard yellow/red/blue tier colouring to help them look more distinctive (orange for bronze, white for chrome). I also took the opportunity to give some other tiered components (e.g. pistons, frames) the same treatment to unify things a bit, break up the metallic monotony, and make tin/iron/steel easier to tell apart. 

![image](https://github.com/Deadlock989/IndustrialRevolution/assets/3128845/0e6a0b7a-827d-4755-b45f-00563f12f4f3)

**Comment by Deadlock989** [comment]
I solved my uneasiness about "space coal" by killing it off - mining probes now only return the four main ores plus "comet ice", which melts down to natural gas and water. My thinking was that if cargo transmats are designed to partly replace mining, there needs to be some carbon element, because a big part of rockets and science is steel, and steel needs coke. While I was reading around I noticed that folks in the real world are starting to experiment with a cokeless method of iron reduction to make steel in arc furnaces using hydrogen instead of carbon, so I added an advanced steel production method to the main IR3 mod. That means the natural gas returned from comet ice could be broken into hydrogen and put into steel production, meaning space mining doesn't also need to return solid carbon.

I don't think this add-on can or necessarily should replace actual mining but it was nice to close that loophole. As it stands at least some of all the main mining products are returned by cargo transmat - the main metal ores, which means some gravel, silica and sulphur via their washing, plus the natural gas and water. 

I'll release this alongside 3.1.0 and if it's too under- or over-powered it can be tweaked later.
